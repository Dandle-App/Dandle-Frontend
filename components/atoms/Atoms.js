import styled from 'styled-components/native';
import {colors} from '../../styles/styles';
import {View, Text, Image, TouchableOpacity, TextInput} from 'react-native';

// colors
const {background, primary, secondary, textLink,
  textLight, textDark, backgroundLight} = colors;

// Form atoms
export const TextInputField = styled.TextInput`
    background-color: ${backgroundLight};
    borderBottomWidth: 3px;
    borderColor: #d3d3d3;
    padding: 10px;
    paddingTop: 5px;
    marginBottom: 35px;
    color: ${textDark};
`;

// texts
export const StyledInputLabel = styled.Text`
    color: ${primary};
    font-size: 16px;
    padding: 5px;
    padding-bottom: 0px;
    text-align: left;
`;
export const MsgBox = styled.Text`
  text-align: center;
  font-size: 13px;
  padding: 3px;
  color: ${(props) => props.type == 'success' ? 'green' : 'red'};
`;

// buttons and links
export const RightIcon = styled.TouchableOpacity`
    right: 1px;
    width: 35px;
    height: 35px;
    top: 35px;
    position: absolute;
    z-index: 200;
`;
export const TextLink = styled.TouchableOpacity`
    align-items: center;
    color: ${textLink};
    paddingLeft: 5px;
`;

// Views
export const LeftIcon = styled.View`
    z-index: 1;
    position: absolute;
    top: 60%;
    left: 2%;
`;
