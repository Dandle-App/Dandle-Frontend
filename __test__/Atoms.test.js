import React from 'react';
import {cleanup, render, fireEvent, waitFor} from '@testing-library/react-native';
import { toHaveStyle } from "@testing-library/jest-native";
import styled from 'styled-components';
import NavigationContainer from '@react-navigation/native';
import renderer from 'react-test-renderer';
import 'jest-styled-components'
import { MsgBox } from '../components/atoms/Atoms';

afterEach(cleanup);

describe('Atoms', () => {

    it('MsgBox', () =>{
        const tree = renderer.create(
            <div>
            <MsgBox type="success">success</MsgBox>
            <MsgBox type="fail">fail</MsgBox>
            </div>
          ).toJSON();
          expect(tree).toMatchSnapshot();

    });
});