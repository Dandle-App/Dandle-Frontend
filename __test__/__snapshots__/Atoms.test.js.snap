// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atoms MsgBox 1`] = `
<div>
  <Text
    style={
      Array [
        Object {
          "color": "green",
          "fontSize": 13,
          "paddingBottom": 3,
          "paddingLeft": 3,
          "paddingRight": 3,
          "paddingTop": 3,
          "textAlign": "center",
        },
      ]
    }
    type="success"
  >
    success
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "red",
          "fontSize": 13,
          "paddingBottom": 3,
          "paddingLeft": 3,
          "paddingRight": 3,
          "paddingTop": 3,
          "textAlign": "center",
        },
      ]
    }
    type="fail"
  >
    fail
  </Text>
</div>
`;
